checker/src/octagon_domain.rs

```diff
@@ -0,0 +1,238 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+//
+
```

Comment:
It would be good to add a short description of this intuition behind the Octagon domain here.